sippy-ng/src/component_readiness/CompReadyVars.js

```diff
@@ -192,38 +178,69 @@ export const CompReadyVarsProvider = ({ children }) => {
-      includeVariantsCheckedItems[variant].forEach((value) => {
-        param.push(variant + ':' + value)
-      })
-    })
-    return param
-  }
+  /********************************************************
+   * All things related to parameter selection for variants
+   ******************************************************** */
 
-  const convertParamToIncludeVariantsCheckedItems = (includeVariantParam) => {
-    let includeVariants = {}
-    includeVariantParam.forEach((variant) => {
-      let kv = variant.split(':')
-      if (kv.length == 2) {
-        if (kv[0] in includeVariants) {
-          includeVariants[kv[0]].push(kv[1])
-        } else {
-          includeVariants[kv[0]] = [kv[1]]
-        }
-      }
-    })
-    return includeVariants
-  }
-  const includeVariantsCheckedItems = convertParamToIncludeVariantsCheckedItems(
+  /** URL parameters for variants **/
+  // The variants that have been selected for inclusion in the basis and sample (unless they are in the cross-compare list)
+  const [
+    includeVariantsCheckedItemsParam = [
+      'Architecture:amd64',
+      'FeatureSet:default',
+      'Installer:ipi',
+      'Installer:upi',
+      'Owner:eng',
+      'Platform:aws',
+      'Platform:azure',
+      'Platform:gcp',
+      'Platform:metal',
+      'Platform:vsphere',
+      'Topology:ha',
+    ],
+    setIncludeVariantsCheckedItemsParam,
+  ] = useQueryParam('includeVariant', ArrayParam)
+  // The list of variant groups (e.g. "Architecture") that have been selected for cross-variant comparison
+  const [variantCrossCompareParam = [], setVariantCrossCompareParam] =
+    useQueryParam('variantCrossCompare', ArrayParam)
+  // The list of individual variants (e.g. "Architecture:arm64") that are checked for cross-variant comparison
+  const [
+    compareVariantsCheckedItemsParam = [],
+    setCompareVariantsCheckedItemsParam,
+  ] = useQueryParam('compareVariant', ArrayParam)
+
+  /** some state variables and related handlers for managing the display of variants in the UI **/
+  // The grouped variants that have been selected for inclusion in the basis and sample (unless they are in the cross-compare list)
+  const includeVariantsCheckedItems = convertParamToVariantItems(
     includeVariantsCheckedItemsParam
   )
   const replaceIncludeVariantsCheckedItems = (variant, checkedItems) => {
     includeVariantsCheckedItems[variant] = checkedItems
   }
+  // The grouped variants that have been selected for cross-comparison in the sample
+  const compareVariantsCheckedItems = convertParamToVariantItems(
+    compareVariantsCheckedItemsParam
+  )
+  const replaceCompareVariantsCheckedItems = (variant, checkedItems) => {
+    compareVariantsCheckedItems[variant] = checkedItems
+  }
+  // This is the list of variant groups (e.g. "Architecture") that have been selected for cross-variant comparison
+  const [variantCrossCompare, setVariantCrossCompare] = useState(
+    variantCrossCompareParam
+  )
```

Comment:
My js is weak, but why is this required when variantCrossCompareParam already exists?

Reply:
This was what we mentioned yesterday, which I found out the hard way when I had the same reaction... have to keep app state and url state separate. When I tried using `variantCrossCompareParam` for app state, every time I checked a box it would update the url params and thus the page. So instead, in-app state keeps track of the UI, and then when you `Generate Report` the Param gets updated.